### ISSUE TITLE
WordPress Microsoft Azure Site Kit

### ISSUE THREAD

**Issue by XVII** [10up/windows-azure-storage#127]
**Is your enhancement related to a problem? Please describe.**

Was looking at options for notifying Azure Front Door of content changes (running purge) and thought it'd be a good opportunity for a plugin.

**Describe the solution you'd like**

Rather than make a new plugin, perhaps the scope of the Azure Media plugin can be expanded to an Azure Site Kit which does a few Azure related tasks.

* Notify Azure Front Door of content changes
* Suggest recommended hosting/environment parameters for WordPress
* Use service principal to authenticate to Azure resources
* Azure AD authentication for users (merge in existing plugin -- needs some TLC)
* Set up / monitor cron via automation or web jobs?

**Comment by jeffpaul** [comment]
@ShadowXVII thanks for the feedback!  Do you know of other Azure-specific plugins that provide any of this functionality?

**Comment by XVII** [comment]
No, just the authentication to Azure AD which isn't even registered in WordPress directory yet.

https://github.com/psignoret/aad-sso-wordpress

It's had bursts of development but a more cohesive Azure kit would be useful.

CloudFlare have a WordPress plugin that notifies their services of outdated content, which is where the Front Door idea came from.

**Comment by XVII** [comment]
Oh and another feature, CDN upload for CSS/JS assets as well (not just images). Not sure the best way to manage that -- perhaps through another plugin?
